website/blog/2024-07-08-building-your-semantic-layer-in-pieces.md

```diff
@@ -0,0 +1,96 @@
+authors: [gwen_windflower]
+tags: [analytics craft]
+hide_table_of_contents: false
+date: 2024-07-09
+is_featured: true
+---
+
+The dbt Semantic Layer is founded on the idea that data transformation should be both _flexible_ — allowing for on-the-fly aggregations grouped and filtered by definable dimensions — as well as _version-controlled and tested_ — like any other codebase you should have confidence that your transformations express your organization’s business logic correctly. Historically, you had to choose between these options, but the dbt Semantic Layer brings them together. This has required new paradigms for _how_ you express your transformations though.
+
+<!-- truncate -->
+
+Because of this, we’ve noticed when talking to dbt users that they _want_ to adopt the Semantic Layer, but feel daunted by the idea of migrating their transformations to this new paradigm. The good news is that you do _not_ need to make a huge one-time migration.
+
+We’re here to discuss another way, building a Semantic Layer in pieces, and our goal is to make sure you derive increased leverage and velocity from each step on your journey. If you’re eager to start building, but like most busy analytics engineers have limited bandwidth, this one is especially for you.
+
+## System of a Noun: deciding what happens where
+
+When you’re using the dbt Semantic Layer, you want to _minimize_ _the modeling that exists outside of dbt_. Eliminate it completely if you can. Why?
+
+- It’s **duplicative, patchy, and confusing** as discussed above.
+- It’s **less powerful**.
+- You **can’t** **test** it.
+- Depending on the tool, oftentimes you **can’t** **version control** it.
+
+What you want is a unified development flow that handles **normalized transformation in dbt models** and **dynamic denormalization in the dbt Semantic Layer**.
+
+:::info
+🏎️ **The Semantic Layer is a denormalization engine.** dbt transforms your data into clean, normalized marts. The dbt Semantic Layer is a denormalization engine that dynamically connects and molds these building blocks into the maximum amount of shapes available _dynamically_.
+:::
+
+This enables a more **flexible consumption layer**, meaning downstream tools (like AI or dashboards) can sit as directly on top of Semantic Layer-generated artifacts and APIs as possible, and focus on what makes them shine instead of being burdened by basic dynamic modeling and aggregation tasks. Any tool-specific constructs should typically operate as close to **transparent pass-throughs** as you can make them, primarily serving to surface metrics and dimensions from the Semantic Layer in your downstream tool. There may be exceptions of course, but as a general guiding principle this gets you the most dynamic denormalization ability, and thus value, from your Semantic Layer code.
+
+So now we’ve established the system, let’s dig into the _plan_ for how we can get there iteratively.
+
+## The plan: towards iterative velocity
```

Comment:
did you meant to have a colon there?
```suggestion
## The plan towards iterative velocity
```

Reply:
@nataliefiann - if the colon is meant to be there - maybe capitalize the T e.g ## The plan: Towards iterative velocity